test/e2e/operator_test.go

```diff
@@ -228,3 +252,195 @@ func setImage(deployment *appsv1.Deployment, image string) {
+
+	// Create the upstream resolver Service and get the ClusterIP.
+	upstreamSvc := upstreamService(upstreamPodName, upstreamPodNS)
+	if err := cl.Create(context.TODO(), upstreamSvc); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+	}
+	defer func() {
+		if err := cl.Delete(context.TODO(), upstreamSvc); err != nil {
+			t.Fatalf("failed to delete service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+		}
+	}()
+	if err := cl.Get(context.TODO(), types.NamespacedName{upstreamSvc.Namespace, upstreamSvc.Name}, upstreamSvc); err != nil {
+		t.Fatalf("failed to get service %s/%s: %v", upstreamSvc.Namespace, upstreamSvc.Name, err)
+	}
+	upstreamIP := upstreamSvc.Spec.ClusterIP
+
+	// Update cluster DNS forwarding with the upstream resolver's Service IP address.
+	defaultDNS := &operatorv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "default"}, defaultDNS); err != nil {
+		t.Fatalf("failed to get default dns: %v", err)
+	}
+	upstream := operatorv1.Server{
+		Name:  "test",
+		Zones: []string{"foo.com"},
+		ForwardPlugin: operatorv1.ForwardPlugin{
+			Upstreams: []string{upstreamIP},
+		},
+	}
+	defaultDNS.Spec.Servers = []operatorv1.Server{upstream}
+	if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+		t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+	}
+	defer func() {
+		defaultDNS = &operatorv1.DNS{}
+		if err := cl.Get(context.TODO(), types.NamespacedName{Name: "default"}, defaultDNS); err != nil {
+			t.Fatalf("failed to get default dns: %v", err)
+		}
+		if len(defaultDNS.Spec.Servers) != 0 {
+			defaultDNS.Spec = operatorv1.DNSSpec{}
+			if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+				t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+			}
+		}
+	}()
+
+	// Verify that the Corefile of DNS DaemonSet pods have been updated.
+	dnsDaemonSet := &appsv1.DaemonSet{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{"openshift-dns", "dns-default"}, dnsDaemonSet); err != nil {
+		t.Fatalf("failed to get daemonset %s/%s: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	selector, err := metav1.LabelSelectorAsSelector(dnsDaemonSet.Spec.Selector)
+	if err != nil {
+		t.Fatalf("daemonset %s/%s has invalid spec.selector: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	defaultDNSPods := &corev1.PodList{}
+	if err := cl.List(context.TODO(), defaultDNSPods, client.MatchingLabelsSelector{Selector: selector}); err != nil {
+		t.Fatalf("failed to list pods for dns daemonset %s/%s: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	catCmd := []string{"cat", "/etc/coredns/Corefile"}
+	dnsContainer := "dns"
+	for _, pod := range defaultDNSPods.Items {
+		if err := lookForStringInPodExec(pod.Namespace, pod.Name, dnsContainer, catCmd, upstreamIP, 2*time.Minute); err != nil {
+			t.Fatalf("failed to find %s in %s of pod %s/%s: %v", upstreamIP, catCmd[1], pod.Namespace, pod.Name, err)
```

Comment:
This is a 2 minute timeout times 6 pods, meaning it could take up to 12 minutes of the update somehow failed and none of the pods has the change.  Would it be reasonable to wait, say, 2 minutes for the daemonset to roll out and then check each pod with a shorter timeout?

Reply:
yes, let me try that.